packages/common/src/utils.ts

```diff
@@ -397,3 +395,116 @@ export function intToBigInt(value: IntegerType, signed: boolean): bigint {
 export function with0x(value: string): string {
   return !value.startsWith('0x') ? `0x${value}` : value;
 }
+
+export const intToHexString = (integer: IntegerType, lengthBytes = 8): string => {
+  const value = typeof integer === 'bigint' ? integer : intToBigInt(integer, false);
+  return value.toString(16).padStart(lengthBytes * 2, '0');
+};
+
+export function hexToBytes(hex: string): Uint8Array {
+  if (typeof hex !== 'string')
+    throw new TypeError('hexToBytes: expected string, got ' + typeof hex);
+  if (hex.length % 2)
+    throw new Error(`hexToBytes: received invalid unpadded hex, got: ${hex.length}`);
+  const array = new Uint8Array(hex.length / 2);
+  for (let i = 0; i < array.length; i++) {
+    const j = i * 2;
+    array[i] = Number.parseInt(hex.slice(j, j + 2), 16);
+  }
+  return array;
+}
+
+const byteToHexCache: string[] = new Array(0xff);
+
+for (let n = 0; n <= 0xff; ++n) {
+  byteToHexCache[n] = n.toString(16).padStart(2, '0');
+}
+
+export function bytesToHex(uint8a: Uint8Array) {
+  const hexOctets = new Array(uint8a.length);
+  for (let i = 0; i < uint8a.length; ++i) hexOctets[i] = byteToHexCache[uint8a[i]];
+  return hexOctets.join('');
+}
+
+export class BN {
```

Comment:
Is a goal of this PR to avoid breaking changes? If so, this will prevent other libraries from passing a `bn.js` lib instance of `BN` because the class types are different, and the functionality is different.

If we don't care about breaking changes, then I'm not sure why we'd try to replicate the bn.js structure when everything can use a `bigint` instance, and call any of these helper functions when needed.